src/server_manager/web_app/digitalocean_server.ts

```diff
@@ -160,27 +135,45 @@ class DigitaloceanServer extends ShadowboxServer implements server.ManagedServer
 
     // Periodically refresh the droplet info then try to update the install
     // state again.
-    const intervalId = setInterval(() => {
+    const intervalId = setInterval(async () => {
       // Check if install state is already known, so we don't make an unnecessary
       // request to fetch droplet info.
       if (this.installState !== InstallState.UNKNOWN) {
         clearInterval(intervalId);
         return;
       }
-      this.refreshDropletInfo().then(() => {
-        updateInstallState();
-        // Immediately clear the interval if the installState is known to prevent
-        // race conditions due to setInterval firing async.
-        if (this.installState !== InstallState.UNKNOWN) {
-          clearInterval(intervalId);
-          return;
-        }
-      });
+      try {
+        await this.refreshDropletInfo();
+      } catch (error) {
+        console.log('Failed to get droplet info', error);
+        this.setInstallState(InstallState.ERROR);
+        clearInterval(intervalId);
+        return;
+      }
+      updateInstallState();
+      // Immediately clear the interval if the installState is known to prevent
+      // race conditions due to setInterval firing async.
+      if (this.installState !== InstallState.UNKNOWN) {
+        clearInterval(intervalId);
+        return;
+      }
       // Note, if there is an error refreshing the droplet, we should just
       // try again, as there may be an intermittent network issue.
     }, 3000);
   }
 
+  private setInstallState(installState: InstallState) {
+    if (this.installState !== InstallState.UNKNOWN) {
```

Comment:
I think this may be too restrictive because the server can go from an `SUCCESS` state to `DELETED`.

Reply:
This is preserving the original behavior. The idea is really just to determine the result of the installation. So it can't really go from SUCCESS to DELETED.

Reply:
👍